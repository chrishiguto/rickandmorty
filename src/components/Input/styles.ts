import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: transparent;
    outline: none;
    border: 2px solid ${theme.colors.white};
    border-radius: ${theme.border.radius};
    transition: box-shadow ${theme.transition.default};

    &:focus-within {
      box-shadow: 0 0 0 0.3rem ${theme.colors.yellow};
    }
  `}
`

export const Input = styled.input`
  ${({ theme }) => css`
    border: none;
    outline: none;
    background: transparent;

    width: 100%;
    padding: ${theme.spacings.xxsmall} calc(${theme.spacings.xsmall} - 0.2rem);
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.white};

    &&::placeholder {
      color: ${theme.colors.lightGray};
    }
  `}
`
