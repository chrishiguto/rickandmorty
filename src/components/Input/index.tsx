import { InputHTMLAttributes, forwardRef } from 'react'

import * as S from './styles'

export type InputProps = InputHTMLAttributes<HTMLInputElement>

const Input: React.ForwardRefRenderFunction<HTMLInputElement, InputProps> = (
  { ...props },
  ref
) => (
  <S.Wrapper>
    <S.Input ref={ref} {...props} />
  </S.Wrapper>
)

export default forwardRef(Input)
