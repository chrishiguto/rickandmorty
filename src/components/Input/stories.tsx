import { Story, Meta } from '@storybook/react/types-6-0'
import Input, { InputProps } from '.'

export default {
  title: 'Input',
  component: Input,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story<InputProps> = (args) => <Input {...args} />

Default.args = {
  placeholder: 'Select an item...'
}
