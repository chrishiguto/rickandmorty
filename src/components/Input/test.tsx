import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Input from '.'

describe('<Input />', () => {
  it('should render the input', () => {
    const { container } = renderWithTheme(
      <Input placeholder="Input placeholder" />
    )

    expect(
      screen.getByPlaceholderText(/input placeholder/i)
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
