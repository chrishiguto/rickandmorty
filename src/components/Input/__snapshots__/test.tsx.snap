// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the input 1`] = `
.c0 {
  background: transparent;
  outline: none;
  border: 2px solid #ffffff;
  border-radius: 0.8rem;
  -webkit-transition: box-shadow 0.3s cubic-bezier(0.4,0,0.2,1);
  transition: box-shadow 0.3s cubic-bezier(0.4,0,0.2,1);
}

.c0:focus-within {
  box-shadow: 0 0 0 0.3rem #cbd736;
}

.c1 {
  border: none;
  outline: none;
  background: transparent;
  width: 100%;
  padding: 0.8rem calc(1.6rem - 0.2rem);
  font-size: 1.6rem;
  font-weight: 700;
  color: #ffffff;
}

.c1.c1::-webkit-input-placeholder {
  color: #D3D3D3;
}

.c1.c1::-moz-placeholder {
  color: #D3D3D3;
}

.c1.c1:-ms-input-placeholder {
  color: #D3D3D3;
}

.c1.c1::placeholder {
  color: #D3D3D3;
}

<div
  class="c0"
>
  <input
    class="c1"
    placeholder="Input placeholder"
  />
</div>
`
